### PR TITLE
ci: only retry "kubectl create" if objects are missing

### PR DESCRIPTION
**Pull request by nixpanic**
There can be spurious failures in the CI when running kubectl create. On
occasion, the command returns with an error, but the api-server did
receive and process the request. This causes a 2nd create action to fail
with messages like:

    cephcluster.ceph.rook.io/my-cluster created
    Error from server: error when creating "/tmp/tmp.Ur1ZPG85o9/cluster-test.yaml": etcdserver: request timed out
    Error from server (AlreadyExists): error when creating "/tmp/tmp.Ur1ZPG85o9/cluster-test.yaml": configmaps "rook-config-override" already exists
    Error from server (AlreadyExists): error when creating "/tmp/tmp.Ur1ZPG85o9/cluster-test.yaml": cephclusters.ceph.rook.io "my-cluster" already exists
    Error from server (AlreadyExists): error when creating "/tmp/tmp.Ur1ZPG85o9/cluster-test.yaml": configmaps "rook-config-override" already exists
    Error from server (AlreadyExists): error when creating "/tmp/tmp.Ur1ZPG85o9/cluster-test.yaml": cephclusters.ceph.rook.io "my-cluster" already exists
    Error from server (AlreadyExists): error when creating "/tmp/tmp.Ur1ZPG85o9/cluster-test.yaml": configmaps "rook-config-override" already exists
    Error from server (AlreadyExists): error when creating "/tmp/tmp.Ur1ZPG85o9/cluster-test.yaml": cephclusters.ceph.rook.io "my-cluster" already exists
    Error from server (AlreadyExists): error when creating "/tmp/tmp.Ur1ZPG85o9/cluster-test.yaml": configmaps "rook-config-override" already exists
    Error from server (AlreadyExists): error when creating "/tmp/tmp.Ur1ZPG85o9/cluster-test.yaml": cephclusters.ceph.rook.io "my-cluster" already exists

By handling the create action differently, and checking for the
AlreadyExists word in the stderr output, it is possible to detect
repeated creates that are not needed.

### CODE REVIEW

File: scripts/rook.sh [hunk 1]
```diff
@@ -13,18 +13,39 @@ rook_version() {
...
 
 kubectl_retry() {
-    local retries=0
+    local retries=0 action="${1}" ret=0 stdout stderr
+    shift
 
-    while ! kubectl "${@}"
+    # temporary files for kubectl output
+    stdout=$(mktemp rook-kubectl-stdout.XXXXXXXX)
+    stderr=$(mktemp rook-kubectl-stdout.XXXXXXXX)
```
Comment (Madhu-1) [inline scripts/rook.sh:21]:
```suggestion
    stderr=$(mktemp rook-kubectl-stderr.XXXXXXXX)
```

Comment (nixpanic) [reply scripts/rook.sh:21]: duh!

**Comment by nixpanic** [comment]
and, before exiting the function, `stdout` and `stderr` need to be written to `/dev/stdout` and `/dev/stderr` so that consumers can access the output

**Comment by Madhu-1** [comment]
@mergifyio rebase

**Pull request merged**